[pybinding] Add mapping from C++ program::verification to Python

larryliu0820 opened the pull request:
Summary: As titled. This enables
`portable_lib._load_for_executorch[_from_buffer]` to accept `Program::Verification` argument.

See added test, now we can do something like:

```
from executorch.extension.pybindings.portable_lib import Verification
module = load_fn(
  exported_program.buffer,
  enable_etdump=False,
  debug_buffer_size=0,
  program_verification=Verification.Minimal,
)
```

Test Plan: See unit test

Reviewers:

Subscribers:

Tasks:

Tags:

dbort commented on extension/pybindings/pybindings.cpp:
```diff
@@ -168,13 +168,14 @@ class Module final {
       std::unique_ptr<ETDumpGen> tracer = nullptr,
-      size_t debug_buffer_size = 0)
+      size_t debug_buffer_size = 0,
+      Program::Verification load_program_config =
```
"config" is vague; `load_program_verification` or just `program_verification` would be more descriptive and precise. Same for other params with the same name.

dbort commented on extension/pybindings/pybindings.cpp:
```diff
@@ -944,19 +966,29 @@ PYBIND11_MODULE(EXECUTORCH_PYTHON_MODULE_NAME, m) {
       call_guard<py::scoped_ostream_redirect, py::scoped_estream_redirect>();
+  
+  // Bind the verification enum to python.
+  py::enum_<Program::Verification>(m, "Verification")
```
Please add an entry to the .pyi file to describe this new type

dbort commented on extension/pybindings/pybindings.cpp:
```diff
@@ -944,19 +966,29 @@ PYBIND11_MODULE(EXECUTORCH_PYTHON_MODULE_NAME, m) {
       py::arg("path"),
       py::arg("enable_etdump") = false,
       py::arg("debug_buffer_size") = 0,
+      py::arg("load_program_config") =
```
Please update the pyi file to reflect these new parameters https://github.com/pytorch/executorch/blob/main/extension/pybindings/pybindings.pyi#L151

dbort reviewed (changes requested):
Thanks for adding and testing this! Looks good structurally but I'd like to change the names and reflect the changes in the pyi file

dbort reviewed (approved):
Thanks for changing the param name and updating the .pyi file!

larryliu0820 committed:
Add mapping from C++ program::verification to Python (#5915)

Summary:
As titled. This enables
`portable_lib._load_for_executorch[_from_buffer]` to accept `Program::Verification` argument.

See added test, now we can do something like:

```
from executorch.extension.pybindings.portable_lib import Verification
module = load_fn(
  exported_program.buffer,
  enable_etdump=False,
  debug_buffer_size=0,
  program_verification=Verification.Minimal,
)
```


Test Plan: See unit test

Reviewed By: dbort

Differential Revision: D63987538

Pulled By: larryliu0820

larryliu0820 commented:
@pytorchbot cherry-pick --onto release/0.4 -c fixnewfeature
